aten/src/ATen/native/mps/operations/Attention.mm

```diff
@@ -19,6 +19,17 @@
 namespace at {
 namespace native {
 
+// expand potential 3d to 4d tensor
+static inline std::tuple<Tensor, bool> ensure_4d(const Tensor& x) {
+    bool squeezed = false;
+    auto t = x;
+    if (x.dim() == 3) {
+        t = x.unsqueeze(0);
+        squeezed = true;
+    }
+    return {std::move(t), squeezed};
```

Comment:
This will avoid unnecessary copies of tensor, wouldn't it?
```suggestion
    if (x.dim() == 3) {
     return {x.unsqueeze(0), true};
   } else {
     return {x, false};
   }
```

Reply:
Or even something like that
```
  const auto is3d = x.dim() == 3;
  return {is3d ? x.unsqueeze(0): x, !is3d};